Framework/DataHandling/src/MultiPeriodLoadMuonStrategy.cpp

```diff
@@ -0,0 +1,138 @@
+    API::WorkspaceGroup_sptr workspace)
+    : LoadMuonStrategy(g_log, filename), m_entry(entry),
+      m_workspaceGroup(std::move(workspace)),
+      m_detectors(getLoadedDetectors()) {}
+
+/**
+ * Loads Muon specific logs into each of the workspaces in the workspace group.
+ * These are logs which are not loaded by LoadISISNexus
+ */
+void MultiPeriodLoadMuonStrategy::loadMuonLogData() {
+
+  auto sampleInformation =
+      LoadMuonNexusV2Helper::loadSampleInformationFromNexus(m_entry);
+  std::string mainFieldDirection =
+      LoadMuonNexusV2Helper::loadMainFieldDirectionFromNexus(m_entry);
+  double firstGoodData =
+      LoadMuonNexusV2Helper::loadFirstGoodDataFromNexus(m_entry);
+
+  for (int i = 0; i < m_workspaceGroup->getNumberOfEntries(); ++i) {
+    auto workspace =
+        std::dynamic_pointer_cast<Workspace2D>(m_workspaceGroup->getItem(i));
+    auto &run = workspace->mutableRun();
+    run.addProperty("main_field_direction", mainFieldDirection);
+    run.addProperty("FirstGoodData", firstGoodData);
+    run.addProperty("sample_temp", sampleInformation.temperature);
+    run.addProperty("sample_magn_field", sampleInformation.magneticField);
+  }
+}
+/**
+ * Loads the good frames data into each of the stored workspace objects
+ */
+void MultiPeriodLoadMuonStrategy::loadGoodFrames() {
+  NXInt goodframes = LoadMuonNexusV2Helper::loadGoodFramesDataFromNexus(
+      m_entry, MULTIPERIODSLOADED);
+  for (int i = 0; i < m_workspaceGroup->getNumberOfEntries(); ++i) {
+    auto workspace =
+        std::dynamic_pointer_cast<Workspace2D>(m_workspaceGroup->getItem(i));
+    auto &run = workspace->mutableRun();
+    run.removeProperty(GOODFRAMESPROP);
+    run.addProperty(GOODFRAMESPROP, goodframes[i]);
+  }
+}
+
+/**
+ * Loads detector grouping.
+ * If no entry in NeXus file for grouping, load it from the IDF.
+ * stored
+ * @returns :: TableWorkspace group containing each of the grouping tables
+ */
+Workspace_sptr MultiPeriodLoadMuonStrategy::loadDetectorGrouping() const {
+  // Each period could in theory have its own grouping, which is reflected in
+  // the nexus file which has (periods*numDetectors) entries in the Nexus
+  // grouping entry.
+  WorkspaceGroup_sptr tableGroup = std::make_shared<WorkspaceGroup>();
+  for (int i = 0; i < m_workspaceGroup->getNumberOfEntries(); ++i) {
+    int periodNumber = i + 1;
+    auto grouping = LoadMuonNexusV2Helper::loadDetectorGroupingFromNexus(
+        m_entry, m_detectors, MULTIPERIODSLOADED, periodNumber);
+    TableWorkspace_sptr table =
+        createDetectorGroupingTable(m_detectors, grouping);
+    // if any of the tables are empty we'll load grouping from the IDF
+    if (table->rowCount() == 0) {
+      m_logger.notice("Loading grouping information from IDF");
+      return LoadMuonNexusV2Helper::loadDefaultDetectorGrouping(
+          std::dynamic_pointer_cast<Workspace2D>(m_workspaceGroup->getItem(i)));
+    } else {
```

Comment:
This else can be dropped since we return in the if and it removes a level of indentation